src/integer_mod_q/z_q/distance.rs

```diff
@@ -0,0 +1,248 @@
+
+//! This module contains the implementation of the [`Distance`] trait for [`Zq`].
+
+use super::Zq;
+use crate::{error::MathError, integer::Z, traits::Distance};
+
+impl Zq {
+    /// Computes the absolute distance between two [`Zq`] instances. It returns the minimum
+    /// distance across boundaries, i.e. `distance(1 mod 7, 6 mod 7) = 2` as `6 == -2 mod 7`.
+    ///
+    /// Parameters:
+    /// - `other`: specifies one of the [`Zq`] values whose distance
+    /// is calculated to `self`
+    ///
+    /// Returns the absolute minimum distance between the two given values as a new
+    /// [`Z`] instance or a [`MathError`] if the moduli mismatch.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::{
+    ///     integer::Z,
+    ///     integer_mod_q::Zq,
+    /// };
+    ///
+    /// let a = Zq::try_from((-1, 13)).unwrap();
+    /// let b = Zq::try_from((5, 13)).unwrap();
+    ///
+    /// let distance = a.distance_safe(&b).unwrap();
+    ///
+    /// assert_eq!(Z::from(6), distance);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type
+    /// [`MismatchingModulus`](MathError::MismatchingModulus) if the
+    /// provided moduli differ.
+    pub fn distance_safe(&self, other: &Zq) -> Result<Z, MathError> {
+        if self.modulus != other.modulus {
+            return Err(MathError::MismatchingModulus(format!(
+                "To compute the distance of two Zq elements, their moduli need to match. 
+                The provided once were {} and {}.",
+                self.modulus, other.modulus,
+            )));
+        }
+
+        let distance_direct = self.value.distance(&other.value);
+        let distance_across_modulus = Z::from(self.modulus.clone()) - &distance_direct;
+
+        if distance_direct < distance_across_modulus {
+            Ok(distance_direct)
+        } else {
+            Ok(distance_across_modulus)
+        }
+        // TODO: use `min` once Ord is implemented for Z
+    }
+}
+
+impl Distance<&Zq> for Zq {
+    type Output = Z;
+
+    /// Computes the absolute distance between two [`Zq`] instances. It returns the minimum
+    /// distance across boundaries, i.e. `distance(1 mod 7, 6 mod 7) = 2` as `6 == -2 mod 7`.
+    ///
+    /// Parameters:
+    /// - `other`: specifies one of the [`Zq`] values whose distance
+    /// is calculated to `self`
+    ///
+    /// Returns the absolute minimum distance between the two given values as a new
+    /// [`Z`] instance or a [`MathError`] if the moduli mismatch.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::{
+    ///     integer::Z,
+    ///     integer_mod_q::Zq,
+    ///     traits::*,
+    /// };
+    ///
+    /// let a = Zq::try_from((-1, 13)).unwrap();
+    /// let b = Zq::try_from((5, 13)).unwrap();
+    ///
+    /// let distance = a.distance(&b);
+    ///
+    /// assert_eq!(Z::from(6), distance);
+    /// ```
+    ///
+    /// # Panics ...
+    /// - if the provided moduli mismatch.
```

Comment:
```suggestion
    /// # Errors and Failures
    /// - Panics if the provided moduli mismatch.
```

Reply:
overlooked it in the style guide, but we have to refactor some code, since it sllipped through at some points